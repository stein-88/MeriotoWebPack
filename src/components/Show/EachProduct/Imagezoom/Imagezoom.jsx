import React from 'react'
import defaultimg from '@images/imgindisponivel.jpg'

const Imagezoom = ({ theimage }) => (
    <div className="container">
        <div className="row">
            <div className="col-12 p-0">
                <section className="text-center bg-dark">
                    <img className="w-100 img-fluid" src={theimage || defaultimg} alt="can be anything of your choice" />
                </section>
            </div>
        </div>
    </div>
)

export default Imagezoom
