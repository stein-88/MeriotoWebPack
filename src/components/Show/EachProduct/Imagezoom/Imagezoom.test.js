import React from 'react'
import renderer from 'react-test-renderer'
import Imagezoom from './Imagezoom'

describe('Test on Imagezoom Component', () => {
    test('Should have an default image when do not pass props', () => {
        const compImagezoom = renderer.create(<Imagezoom />)
        const rootLink = compImagezoom.root.findAllByProps({ src: 'test-file-stub' })
        expect(rootLink).toHaveLength(1)
    })
    test('Should have an image when we pass the props', () => {
        const compImage = renderer.create(<Imagezoom theimage={'linkaq'} />)
        const rootImage = compImage.root.findAllByProps({ className: 'w-100 img-fluid' })
        const rootLink = compImage.root.findAllByProps({ src: 'linkaq' })
        const jsonImagezoom = compImage.toJSON()
        expect(rootImage).toHaveLength(1)
        expect(rootLink).toHaveLength(1)
        expect(jsonImagezoom).toMatchSnapshot()
    })
})