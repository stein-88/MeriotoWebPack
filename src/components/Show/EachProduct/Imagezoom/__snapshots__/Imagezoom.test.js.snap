// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on Imagezoom Component Should have an image when we pass the props 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-12 p-0"
    >
      <section
        className="text-center bg-dark"
      >
        <img
          alt="can be anything of your choice"
          className="w-100 img-fluid"
          src="linkaq"
        />
      </section>
    </div>
  </div>
</div>
`;
